js/js.dce/src/org/jetbrains/kotlin/js/dce/ReachabilityTracker.kt

```diff
@@ -19,19 +19,23 @@ package org.jetbrains.kotlin.js.dce
 import org.jetbrains.kotlin.js.backend.ast.*
 import org.jetbrains.kotlin.js.dce.Context.Node
 import org.jetbrains.kotlin.js.inline.util.collectLocalVariables
+import java.util.concurrent.atomic.AtomicInteger
 
 class ReachabilityTracker(
         private val context: Context,
         private val analysisResult: AnalysisResult,
+        private val collectOnlyRootNodes: Boolean,
         private val logConsumer: (DCELogLevel, String) -> Unit
 ) : RecursiveJsVisitor() {
     companion object {
         private val CALL_FUNCTIONS = setOf("call", "apply")
+        private val NODE_MARKER = AtomicInteger(0)
```

Comment:
Why atomic?

Reply:
Replaced `tag` with boolean flag and also this code